src/server.js

```diff
@@ -58,7 +58,7 @@ class Client extends EventEmitter {
   }
 
   initializeSession(token, data = {}) {
-    if (this.session) sessions.delete(this.session.token);
+    if (this.session) this.finalizeSession(token);
```

Comment:
```suggestion
    this.finalizeSession();
```
- We need no `token` parameter for `finalizeSession` call because it should finalize current one.
- We do not even need `if`-statement here.